core/domain/services/registration/form/v1/RegFormQuestionFactory.php

```diff
@@ -0,0 +1,122 @@
+<?php
+
+namespace EventEspresso\core\domain\services\registration\form\v1;
+
+use EE_Answer;
+use EE_Error;
+use EE_Form_Input_Base;
+use EE_Question;
+use EE_Registration;
+use EEM_Answer;
+use EventEspresso\core\exceptions\InvalidDataTypeException;
+use EventEspresso\core\exceptions\InvalidInterfaceException;
+use InvalidArgumentException;
+use ReflectionException;
+
+class RegFormQuestionFactory
+{
+    /**
+     * @var array
+     */
+    private $required_questions = [];
+
+
+    /**
+     * @return array
+     */
+    public function requiredQuestions(): array
+    {
+        return $this->required_questions;
+    }
+
+
+    /**
+     * @param EE_Registration $registration
+     * @param EE_Question     $question
+     * @return EE_Form_Input_Base
+     * @throws EE_Error
+     * @throws InvalidArgumentException
+     * @throws InvalidDataTypeException
+     * @throws InvalidInterfaceException
+     * @throws ReflectionException
+     */
+    public function create(EE_Registration $registration, EE_Question $question): EE_Form_Input_Base
+    {
+        // if this question was for an attendee detail, then check for that answer
+        $answer_value = EEM_Answer::instance()->get_attendee_property_answer_value(
+            $registration,
+            $question->system_ID()
+        );
+        $answer       = $answer_value === null
+            ? EEM_Answer::instance()->get_one(
+                [['QST_ID' => $question->ID(), 'REG_ID' => $registration->ID()]]
+            )
+            : null;
+        // if NOT returning to edit an existing registration
+        // OR if this question is for an attendee property
+        // OR we still don't have an EE_Answer object
+        if ($answer_value || ! $answer instanceof EE_Answer || ! $registration->reg_url_link()) {
```

Comment:
TIL `instanceof` can be negated without parenthesis. But the scope of negation looks a bit confusing this way.